app/models/user.rb

```diff
@@ -24,11 +27,19 @@ class User < ApplicationRecord
   scope :nurses, -> { (where(role: [roles[:primary_nurse], roles[:secondary_nurse]])) }
   scope :assignable_users, -> { where("role = ? OR role = ?", "Primary Nurse", "Secondary Nurse").where(facility_id: nil) }
 
-  validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
-  validates :password, presence: true, :on => :create
-  validates :email, uniqueness: true
   validates :phone, uniqueness: true
 
+  attr_accessor :login_id
+
+  def self.find_for_database_authentication(warden_conditions)
+    conditions = warden_conditions.dup
+    if login = conditions.delete(:login_id)
+      where(conditions.to_h).where(["phone = :phone OR lower(email) = :email", { :phone => login.to_i, :email => login.downcase }]).first
```

Comment:
I think this function has no use in codebase. 
If it has some use, I will recommend to use this form. Avoid string interpolation like thing. 

```suggestion
      where(conditions.to_h).where(phone: login.to_i).or(email: login.downcase).first
```

Reply:
📌
https://github.com/coronasafe/arike/blob/eaa3f0439d9b596269e953d27ea463864febd838/app/models/user.rb